### PR TITLE
Upgrade EFA to 1.26.1

### PR DESCRIPTION
**Pull request by himani2411**
The hash of the installer can be retrieved here https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-verify.html

Picked the Commit from Release-3.7 https://github.com/aws/aws-parallelcluster-cookbook/pull/2460

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,15 +57,14 @@ This file is used to list changes made in each version of the AWS ParallelCluste
   - Open MPI: `openmpi40-aws-4.1.5-4`
 - Upgrade Slurm to version 23.02.4.
 - Upgrade ARM PL to version 23.04.1 for Ubuntu 22.04 only.
-- Upgrade Cinc Client to version 18.2.7.
```
Comment (jdeamicis) [inline CHANGELOG.md:60]: It looks like you are right in that Cinc was not upgraded with 3.7.0. Was this a mistake in the writing of the changelog?

File: CHANGELOG.md [hunk 2]
```diff
@@ -57,15 +57,14 @@ This file is used to list changes made in each version of the AWS ParallelCluste
...
-  - selinux-6.1.12 (from selinux-6.0.5)
-  - yum-7.4.13 (from yum-7.4.0)
-  - yum-epel-5.0.2 (from yum-epel-4.5.0)
+    - apt-7.5.14 (from apt-7.4.0)
+    - line-4.5.13 (from line-4.5.2)
+    - openssh-2.11.3 (from openssh-2.10.3)
+    - pyenv-4.2.3 (from pyenv-3.5.1)
+    - selinux-6.1.12 (from selinux-6.0.5)
+    - yum-7.4.13 (from yum-7.4.0)
+    - yum-epel-5.0.2 (from yum-epel-4.5.0)
```
Comment (jdeamicis) [inline CHANGELOG.md]: I guess you have the check yaml pre-commit hook enabled. Could you revert this change in this block please?

File: CHANGELOG.md [hunk 3]
```diff
@@ -362,7 +361,6 @@ This file is used to list changes made in each version of the AWS ParallelCluste
 **ENHANCEMENTS**
 - Add support for multiple users cluster environments by integrating with Active Directory (AD) domains managed via AWS Directory Service.
 - Install NVIDIA drivers and CUDA library for ARM.
-- Build Slurm with `slurmrestd` support.
```
Comment (jdeamicis) [inline CHANGELOG.md]: Why is this line being removed?

**Review by jdeamicis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (himani2411) [reply CHANGELOG.md:60]: I think we didnt change the changelog for develop branch.

File: CHANGELOG.md (hunk 3 above)
Comment (himani2411) [reply CHANGELOG.md]: Its weird that changelog is not consistent throughout the branches. 


This line was not in [release-3.7](https://github.com/aws/aws-parallelcluster-cookbook/blob/release-3.7/CHANGELOG.md) changelog and neither in [release-3.1](https://github.com/aws/aws-parallelcluster-cookbook/blob/release-3.1/CHANGELOG.md)

File: CHANGELOG.md (hunk 2 above)
Comment (himani2411) [reply CHANGELOG.md]: Will change it

File: CHANGELOG.md (hunk 3 above)
Comment (jdeamicis) [reply CHANGELOG.md]: Let's keep this line please. We may consider fixing the 3.7 branch later.

File: CHANGELOG.md (hunk 1 above)
Comment (jdeamicis) [reply CHANGELOG.md:60]: It's fine, let's remove it.

File: CHANGELOG.md (hunk 3 above)
Comment (himani2411) [reply CHANGELOG.md]: I see the change in develop changelog was added in this https://github.com/aws/aws-parallelcluster-cookbook/pull/2446.

but not picked for release-3.7 branch

Comment (jdeamicis) [reply CHANGELOG.md]: Yep, I guess they didn't want to modify all the release branches.

**Review by jdeamicis — APPROVED** [review]
